hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -646,16 +629,31 @@ public Set entrySet(Predicate predicate) {
     @Override
     @SuppressWarnings("unchecked")
     public Collection<V> values(Predicate predicate) {
+        return executePredicate(predicate, IterationType.VALUE, false);
+    }
+
+    // todo: this logic should be moved to the query engine because interpreting the query is not a task
+    // of the MapProxy.
```

Comment:
I have no such plans. The query engine is a pure a server side matter and if the client code needs a cleanup dealing with predicates, it should be done in a different task.